sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosTableSchemaInferrer.scala

```
@@ -117,6 +118,15 @@ private object CosmosTableSchemaInferrer
       SparkUtils.safeOpenConnectionInitCaches(sourceContainer, (msg, e) => logWarning(msg, e))
       val queryOptions = new CosmosQueryRequestOptions()
       queryOptions.setMaxBufferedItemCount(cosmosInferenceConfig.inferSchemaSamplingSize)
+
+      cosmosReadConfig.maxIntegratedCacheStaleness match {
+        case CosmosConstants.invalidDuration =>
```

should throw on invalid duration